services/src/datasets/external/netcdfcf/mod.rs

```
@@ -573,63 +617,101 @@ impl NetCdfCfDataProvider {
 
         let entities = root_group
             .dimension_as_string_array("entities")
-            .context(error::MissingEntities)?;
+            .context(error::MissingEntities)?
+            .into_iter()
+            .enumerate()
+            .map(|(id, name)| NetCdfArrayDataset { id, name })
+            .collect::<Vec<_>>();
 
-        let mut group_deque: VecDeque<GroupPath> = root_group
+        let subgroups = root_group
             .group_names()
-            .into_iter()
-            .map(|name| Ok(GroupPath::new(root_group.open_group(&name)?)))
-            .collect::<Result<VecDeque<_>>>()?;
+            .iter()
+            .map(|name| root_group.open_group(name)?.to_net_cdf_subgroup())
+            .collect::<Result<Vec<_>>>()?;
 
-        let mut datasets = Vec::new();
+        let start = root_group
+            .attribute_as_string("time_coverage_start")
+            .context(error::MissingTimeCoverageStart)?;
+        let end = root_group
+            .attribute_as_string("time_coverage_end")
+            .context(error::MissingTimeCoverageEnd)?;
+        let step = root_group
+            .attribute_as_string("time_coverage_resolution")
+            .context(error::MissingTimeCoverageResolution)?;
 
-        while let Some(group_path) = group_deque.pop_front() {
-            let subgroups = group_path
-                .current
-                .group_names()
-                .into_iter()
-                .map(|name| Ok(group_path.extend(group_path.current.open_group(&name)?)))
-                .collect::<Result<Vec<_>>>()?;
+        let (time_start, time_end, time_step) = parse_time_coverage(&start, &end, &step)?;
+
+        Ok(NetCdfOverview {
+            file_name: file_name.to_string(),
+            title,
+            spatial_reference,
+            subgroups,
+            entities,
+            time: TimeInterval::new(time_start, time_end)
+                .unwrap_or_else(|_| TimeInterval::from(time_start)),
```

I'd rather return an error here.

---

b46b8a4